puma/apps/android/whatsapp/whatsapp.py

```diff
@@ -210,13 +213,23 @@ def send_media(self, directory_name, caption=None, view_once=False, chat: str =
         # Go to gallery
         self.driver.find_element(by=AppiumBy.ID, value="com.whatsapp:id/input_attach_button").click()
         self.driver.find_element(by=AppiumBy.ID, value="com.whatsapp:id/pickfiletype_gallery_holder").click()
-        self.driver.find_element(by=AppiumBy.XPATH,
-                                 value='//android.widget.LinearLayout[@content-desc="Gallery"]').click()
-        directory_tile = f'//android.widget.TextView[@resource-id="com.whatsapp:id/title" and @text="{directory_name}"]'
-        self.swipe_to_find_element(xpath=directory_tile).click()
-
-        sleep(0.5)
-        self.driver.find_element(by=By.CLASS_NAME, value="android.widget.ImageView").click()
+        if self.is_present('//android.widget.LinearLayout[@content-desc="Gallery"]'):
```

Comment:
Maybe add a comment above each of these two cases to explain a bit more?

E.g., what is the first case, when does it occur, and what is the second? I find it hard to understand by just looking at the XPath expressions.

Alternatively, we could extract two methods with meaningful names for each case.